client/src/main/kotlin/io/spine/examples/chatspn/desktop/Client.kt

```diff
@@ -0,0 +1,414 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.chatspn.desktop
+
+import com.google.protobuf.Message
+import io.grpc.ManagedChannelBuilder
+import io.spine.base.EntityColumn
+import io.spine.base.EntityStateField
+import io.spine.base.EventMessage
+import io.spine.base.EventMessageField
+import io.spine.base.Field
+import io.spine.client.Client
+import io.spine.client.ClientRequest
+import io.spine.client.ConnectionConstants.DEFAULT_CLIENT_SERVICE_PORT
+import io.spine.client.EntityStateFilter
+import io.spine.client.EventFilter
+import io.spine.client.OrderBy
+import io.spine.client.QueryFilter
+import io.spine.client.Subscription
+import io.spine.core.UserId
+import io.spine.examples.chatspn.AccountCreationId
+import io.spine.examples.chatspn.ChatId
+import io.spine.examples.chatspn.MessageId
+import io.spine.examples.chatspn.account.UserChats
+import io.spine.examples.chatspn.account.UserProfile
+import io.spine.examples.chatspn.account.command.CreateAccount
+import io.spine.examples.chatspn.account.event.AccountCreated
+import io.spine.examples.chatspn.account.event.AccountNotCreated
+import io.spine.examples.chatspn.chat.ChatPreview
+import io.spine.examples.chatspn.chat.command.CreatePersonalChat
+import io.spine.examples.chatspn.message.MessageView
+import io.spine.examples.chatspn.message.command.SendMessage
+import io.spine.examples.chatspn.message.event.MessageMarkedAsDeleted
+import io.spine.net.EmailAddress
+
+/**
+ * Provides API to communicate with ChatSpn server.
+ */
+public class ClientFacade {
+    public var authenticatedUser: UserProfile? = null
+    private val client: Client
+    private val userChatsSubscriptions = mutableListOf<Subscription>()
+    private val messagesSubscriptions = mutableListOf<Subscription>()
+
+    init {
+        val channel = ManagedChannelBuilder.forAddress(
+            "localhost",
```

Comment:
Can we please include at least some "default" expectations into the class-level documentation?